test/unit/CodeGenerator/PhpParserGenerators/DtoCodeGeneratorTest.php

```diff
@@ -0,0 +1,785 @@
+            ],
+            [
+                'payload' => [
+                    'definition' => self::DEFINITION_BASIC,
+                    'codeGenerator' => [
+                        'languageLevel' => '7.4',
+                        'fullDocs' => false,
+                    ],
+                ],
+                'expected' => self::EXAMPLE_FILE_CONTENT_BASIC_SHORT_DOC_7_4,
+            ],
+            [
+                'payload' => [
+                    'definition' => self::DEFINITION_EXTENDED,
+                    'codeGenerator' => [
+                        'languageLevel' => '7.4',
+                        'fullDocs' => false,
+                    ],
+                ],
+                'expected' => self::EXAMPLE_FILE_CONTENT_EXTENDED_SHORT_DOC_7_4,
+            ],
+            [
+                'payload' => [
+                    'definition' => self::DEFINITION_EMPTY_PROPERTIES,
+                    'codeGenerator' => [
+                        'languageLevel' => '7.4',
+                        'fullDocs' => true,
+                    ],
+                ],
+                'expected' => self::EXAMPLE_FILE_CONTENT_EMPTY_PROPERTIES_FULL_DOC_7_4,
+            ],
+            [
+                'payload' => [
+                    'definition' => self::DEFINITION_BASIC,
+                    'codeGenerator' => [
+                        'languageLevel' => '7.4',
+                        'fullDocs' => true,
+                    ],
+                ],
+                'expected' => self::EXAMPLE_FILE_CONTENT_BASIC_FULL_DOC_7_4,
+            ],
+            [
+                'payload' => [
+                    'definition' => self::DEFINITION_EXTENDED,
+                    'codeGenerator' => [
+                        'languageLevel' => '7.4',
+                        'fullDocs' => true,
+                    ],
+                ],
+                'expected' => self::EXAMPLE_FILE_CONTENT_EXTENDED_FULL_DOC_7_4,
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed[] $payload
+     *
+     * @dataProvider generateProvider
+     */
+    public function testGenerate(array $payload, string $expected): void
+    {
+        $codeGenerator = new DtoCodeGenerator(
+            new BuilderFactory(),
+            new ScalarTypesResolver(),
+            $payload['codeGenerator']['languageLevel'],
+            $payload['codeGenerator']['fullDocs'],
+        );
+
+        $definition = $this->createDtoDefinition($payload['definition']);
+
+        $generatedFileDefinition = $codeGenerator->generate($definition);
+
+        Assert::assertSame($definition, $generatedFileDefinition->getClass());
+        Assert::assertSame($expected, $generatedFileDefinition->getFileContents());
+    }
+
+//    /**
+//     * @deprecated Changed to compare by content.
+//     *
+//     * @param mixed[] $payload
+//     *
+//     * @dataProvider generateProvider
+//     */
+//    public function testGenerate(array $payload): void
```

Comment:
At first there was an idea to make a check for each case, but the implementation seemed excessively complicated.